internal/babysitter/babysitter.go

```diff
@@ -237,19 +285,20 @@ func (b *Babysitter) startColocationGroup(g *group) error {
 			g:          g,
 			subscribed: map[string]bool{},
 		}
-		e, err := envelope.NewEnvelope(b.ctx, wlet, b.dep.App, h)
+		e, err := envelope.NewEnvelope(b.ctx, wlet, b.dep.App)
 		if err != nil {
 			return err
 		}
 		h.envelope = e
-		go func() {
-			if err := e.Serve(b.ctx); err != nil {
-				b.done <- err
-			}
-		}()
 		if err := b.registerReplica(g, e.WeaveletInfo()); err != nil {
 			return err
 		}
+		e.Serve(h)
```

Comment:
I think we need to call `Serve` before `registerReplica`, as `registerReplica` may issue RPCs.